ui/src/main/webapp/app/query/date-range/date-range-component.js

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+angular.module('app').component('dateRange', dateRange());
+
+function dateRange() {
+    return {
+        templateUrl: 'app/query/date-range/date-range.html',
+        controller: DateRangeController,
+        controllerAs: 'ctrl',
+        bindings: {
+            conf: '<'
+        }
+    }
+}
+
+function DateRangeController(time, common) {
+    var vm = this;
+
+    vm.start = null;
+    vm.end = null;
+
+    vm.$onInit = function() {
+        if (!vm.conf) {
+            throw 'Config Error: Date range must be configured';
+        }
+        if (!vm.conf.start || !vm.conf.end) {
+            throw 'Config Error: You must specify the configuration for the start and end date';
+        }
+        if (!vm.conf.start.property || !vm.conf.end.property) {
+            throw 'Config Error: You must specify the start and end property';
+        }
+        if (!vm.conf.start.class || !vm.conf.end.class) {
+            throw 'Config Error: You must specify the class for the start and end';
+        }
+        if (vm.conf.start.unit) {
+            var unit = angular.lowercase(vm.conf.start.unit);
+            if (unit !== 'milliseconds' && unit !== 'microseconds' && unit !== 'seconds') {
+                throw 'Config Error: Unknown start time unit - ' + vm.conf.start.unit + '. Must be one of seconds, milliseconds or microseconds (defaults to milliseconds)';
+            }
+        }
+        if (vm.conf.end.unit) {
+            var unit = angular.lowercase(vm.conf.end.unit);
+            if (unit !== 'milliseconds' && unit !== 'microseconds' && unit !== 'seconds') {
+                throw 'Config Error: Unknown end time unit - ' + vm.conf.end.unit + '. Must be one of seconds, milliseconds or microseconds (defaults to milliseconds)';
+            }
+        }
+
+        var start = time.getStartDate();
+        if (start) {
+            vm.start = convertNumberToDate(start, vm.conf.start.unit);
+        }
+        var end = time.getEndDate();
+        if(end) {
+            vm.end = convertNumberToDate(end, vm.conf.end.unit);
+        }
+    }
+
+    var convertNumberToDate = function(value, unit) {
+
+        if (!unit || angular.lowercase(unit) === 'milliseconds') {
```

Comment:
For consistency, if might be better if these time units matched the time units we use to configure the the date range predicates in Koryphe. Take a look at the Koryphe TimeUnit enum, i.e we should support:
- DAY
- HOUR
- MINUTE
- SECOND
- MILLISECOND
- MICROSECOND